// ngx-materialize/esm5/src/card/card.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { ChangeDetectorRef, Component, Directive, ElementRef, HostBinding, Input, ViewChild, } from '@angular/core';
var MzCardComponent = /** @class */ (function () {
    function MzCardComponent(changeDetectorRef) {
        this.changeDetectorRef = changeDetectorRef;
        this.hasCardAction = true;
        this.hasCardTitle = true;
    }
    /**
     * @return {?}
     */
    MzCardComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        this.hasCardTitle = this.hasTitleTagAndNotEmpty();
        this.hasCardAction = this.hasActionTagAndNotEmpty();
        this.changeDetectorRef.detectChanges();
    };
    /**
     * @return {?}
     */
    MzCardComponent.prototype.hasActionTagAndNotEmpty = /**
     * @return {?}
     */
    function () {
        var /** @type {?} */ cardActionElement = this.cardAction.nativeElement.querySelector('mz-card-action');
        return this.isElementDisplayed(cardActionElement);
    };
    /**
     * @return {?}
     */
    MzCardComponent.prototype.hasTitleTagAndNotEmpty = /**
     * @return {?}
     */
    function () {
        var /** @type {?} */ cardTitleElement = this.cardTitle.nativeElement.querySelector('mz-card-title');
        return this.isElementDisplayed(cardTitleElement);
    };
    /**
     * @param {?} element
     * @return {?}
     */
    MzCardComponent.prototype.isElementDisplayed = /**
     * @param {?} element
     * @return {?}
     */
    function (element) {
        return element && element.innerHTML.trim() !== '';
    };
    MzCardComponent.decorators = [
        { type: Component, args: [{
                    selector: 'mz-card',
                    template: "<div class=\"card-content\">\n\n  <div #cardTitle class=\"card-title\" *ngIf=\"hasCardTitle\">\n    <ng-content select=\"mz-card-title\"></ng-content>\n  </div>\n\n  <ng-content select=\"mz-card-content\"></ng-content>\n</div>\n\n<div #cardAction class=\"card-action\" *ngIf=\"hasCardAction\">\n  <ng-content select=\"mz-card-action\"></ng-content>\n</div>",
                    styles: [":host{display:block}"],
                },] },
    ];
    /** @nocollapse */
    MzCardComponent.ctorParameters = function () { return [
        { type: ChangeDetectorRef, },
    ]; };
    MzCardComponent.propDecorators = {
        "true": [{ type: HostBinding, args: ['class.card',] },],
        "hoverable": [{ type: HostBinding, args: ['class.hoverable',] }, { type: Input },],
        "cardTitle": [{ type: ViewChild, args: ['cardTitle',] },],
        "cardAction": [{ type: ViewChild, args: ['cardAction',] },],
    };
    return MzCardComponent;
}());
export { MzCardComponent };
function MzCardComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    MzCardComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    MzCardComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    MzCardComponent.propDecorators;
    /** @type {?} */
    MzCardComponent.prototype.true;
    /** @type {?} */
    MzCardComponent.prototype.hoverable;
    /** @type {?} */
    MzCardComponent.prototype.cardTitle;
    /** @type {?} */
    MzCardComponent.prototype.cardAction;
    /** @type {?} */
    MzCardComponent.prototype.hasCardAction;
    /** @type {?} */
    MzCardComponent.prototype.hasCardTitle;
    /** @type {?} */
    MzCardComponent.prototype.changeDetectorRef;
}
var MzCardTitleDirective = /** @class */ (function () {
    function MzCardTitleDirective() {
    }
    MzCardTitleDirective.decorators = [
        { type: Directive, args: [{ selector: 'mz-card-title' },] },
    ];
    return MzCardTitleDirective;
}());
export { MzCardTitleDirective };
function MzCardTitleDirective_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    MzCardTitleDirective.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    MzCardTitleDirective.ctorParameters;
}
var MzCardContentDirective = /** @class */ (function () {
    function MzCardContentDirective() {
    }
    MzCardContentDirective.decorators = [
        { type: Directive, args: [{ selector: 'mz-card-content' },] },
    ];
    return MzCardContentDirective;
}());
export { MzCardContentDirective };
function MzCardContentDirective_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    MzCardContentDirective.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    MzCardContentDirective.ctorParameters;
}
var MzCardActionDirective = /** @class */ (function () {
    function MzCardActionDirective() {
    }
    MzCardActionDirective.decorators = [
        { type: Directive, args: [{ selector: 'mz-card-action' },] },
    ];
    return MzCardActionDirective;
}());
export { MzCardActionDirective };
function MzCardActionDirective_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    MzCardActionDirective.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    MzCardActionDirective.ctorParameters;
}
